types/react-navigation/index.d.ts

```diff
@@ -86,7 +87,7 @@ export type NavigationLeafRoute<Params> = {
    * Params passed to this route when navigating to it,
    * e.g. `{ car_id: 123 }` in a route that displays a car.
    */
-  params?: Params,
+  params: Params,
```

Comment:
Yeah, that would certainly be the best case, but as you mentioned, given the decoupling in typings between the local and global contexts, this seems intractable at the time for compilation steps (and anything else before runtime). We'll just leave it up to the user to type their own params at this time, with my suggestion being the form mentioned in my comment above, and copied below:

```
const params: ThisComponentsNavigationParamsInterface = {};
this.props.navigation.navigate(ThisComponentRouteName, params);
```